src/clang-cpp-frontend/clang_cpp_convert.cpp

```diff
@@ -213,21 +213,28 @@ bool clang_cpp_convertert::get_var(const clang::VarDecl &vd, exprt &new_expr)
   return clang_c_convertert::get_var(vd, new_expr);
 }
 
-bool clang_cpp_convertert::get_function(
-  const clang::FunctionDecl &fd,
+bool clang_cpp_convertert::get_method(
+  const clang::CXXMethodDecl &md,
   exprt &new_expr)
 {
   // Only convert instantiated functions/methods not depending on a template parameter
-  if(fd.isDependentContext())
+  if(md.isDependentContext())
+    return false;
+
+  // Don't convert if implicit, unless it's a constructor/destructor/
+  // Copy assignment Operator/Move assignment Operator
+  // A compiler-generated default ctor/dtor is considered implicit, but we have
+  // to parse it.
+  if(
+    md.isImplicit() && !is_ConstructorOrDestructor(md) &&
+    !is_CopyOrMoveOperator(md))
     return false;
 
-  if(clang_c_convertert::get_function(fd, new_expr))
+  if(clang_c_convertert::get_function(md, new_expr))
```

Comment:
implicit conversion of `const clang::CXXMethodDecl &` to `const clang::FunctionDecl &`
@mikhailramalho Is this type of conversion safe in LLVM?